cmd/fs-v1.go

```diff
@@ -20,113 +20,239 @@ import (
 	"crypto/md5"
 	"crypto/sha256"
 	"encoding/hex"
+	"encoding/json"
 	"errors"
 	"fmt"
 	"hash"
 	"io"
-	"path"
+	"io/ioutil"
+	"os"
+	slashpath "path"
+	"path/filepath"
+	"runtime"
 	"sort"
 	"strings"
+	"syscall"
 
+	"github.com/minio/minio/pkg/disk"
+	"github.com/minio/minio/pkg/lock"
 	"github.com/minio/minio/pkg/mimedb"
 )
 
 // fsObjects - Implements fs object layer.
 type fsObjects struct {
-	storage StorageAPI
```

Comment:
The idea behind `locked-posix.go` is to make posix related operation separately and make testable locked posix operations. Now we need to test `FS` for such things. This also makes avoiding repetition of logic.

My proposal is to extend `posix.go` to `locked-posix.go` and have specific methods related to shared fs backend.

Another issue in this PR is it does not handle I/O errors properly like `posix.go`